apps/client/src/routes/dashboards/dashboards-index/hooks/use-partial-update-dashboard-mutation.ts

```diff
@@ -0,0 +1,51 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useIntl } from 'react-intl';
+
+import {
+  updateDashboard,
+  Dashboard,
+  readDashboard,
+  ApiError,
+} from '~/services';
+import { useSendNotification } from '~/hooks/notifications/use-send-notification';
+import { DASHBOARD_SUMMARIES_QUERY_KEY } from '~/data/dashboards';
+
+export function usePartialUpdateDashboardMutation() {
```

Comment:
to better support the frontend table inline and dashboard editing UX flows, no fields are overlapping. It would be better if we split `core` dashboard update api into 2, for [1] updating the summary, [2] definition. Unless the dashboard editing UX also update name/description.

No need to do it in this PR, just want to see what you think.

Reply:
Yeah, agreed. There's no reason why we shouldn't make the optimal API we need instead of doing what's not optimal on the client.